Linear Euler 2D Model (#69)
FluidNumerics/SELF-Fluids: docs/Tutorials/LinearEuler2D/PlaneWaveReflection.md

# Linear Euler 2D - Plane Wave Reflection Tutorial

> This tutorial will walk you through using an example program that uses the `LinearEuler2D` class to run a simulation with the linear Euler equations for an ideal gas in 2-D. This example is configured using the built in structured mesh generator with prescribed boundary conditions on all domain boundaries.

Reviewer: ```suggestion
This tutorial will walk you through using an example program that uses the `LinearEuler2D` class to run a simulation with the linear Euler equations for an ideal gas in 2-D. This example is configured using the built in structured mesh generator with prescribed boundary conditions on north, west, and south boundaries and a no-normal-flow boundary condition on the east boundary.
```

    \rho \\ 
    u \\ 
    v \\ 
    p
    \end{pmatrix}
$$

and

$$
    \overleftrightarrow{f} = 
    \begin{pmatrix}
    \rho_0(u \hat{x} + v \hat{y}) \\
    p \hat{x} \\
    p \hat{y} \\
    \rho_0c^2(u \hat{x} + v \hat{y})
    \end{pmatrix}
$$

$$
    \vec{q} = \vec{0}
$$ 

The variables are defined as follows

* $\rho$ is a density anomaly referenced to the density $\rho_0$
* $u$ and $v$ are the $x$ and $y$ components of the fluid velocity (respectively)
* $p$ is the pressure
* $c$ is the (constant) speed of sound. 

### Model Domain
The physical domain is defined by $\vec{x} \in [0, 1]\times[0,1]$. We use the `StructuredMesh` routine to create a domain with 20 × 20 elements that are dimensioned 0.05 × 0.05 . Model boundaries on the south, north, and west edges of the domain are all tagged with the `SELF_BC_PRESCRIBED` flag to implement prescribed boundary conditions; the east boundary is tagged with `SELF_BC_NONORMALFLOW` to implement no-normal-flow boundary conditions.

Within each element, all variables are approximated by a Lagrange interpolating polynomial of degree 7. The interpolation knots are the Legendre-Gauss points.

### Initial and Boundary Conditions

> The initial and boundary conditions are set using an exact solution. The exact solution is found using the method of images where a no-normal-flow wall is placed at $x=1$ . We define the solution as the sum of an incident wave and a reflecting wave

Reviewer: Not sure if this needs rewording for the no-normal-flow boundary.

```suggestion
The initial and prescribed boundary conditions are set using an exact solution. The exact solution is found using the method of images where a no-normal-flow wall is placed at $x=1$ . We define the solution as the sum of an incident wave and a reflecting wave
```

$$
\vec{s} = \vec{s}_i + \vec{s}_r
$$

where

$$
    \vec{s}_i = 
    \begin{pmatrix}
    \frac{1}{c^2} \\
    \frac{k_x}{c} \\ 
    \frac{k_y}{c} \\ 
    1
    \end{pmatrix} \bar{p} e^{-\left( (\frac{k_x(x-x_0) + k_y(y-y_0) - ct)^2}{L^2} \right)}
$$

is the incident wave, and

$$
    \vec{s}_r = 
    \begin{pmatrix}
    \frac{1}{c^2} \\
    -\frac{k_x}{c} \\ 
    \frac{k_y}{c} \\ 
    1
    \end{pmatrix} \bar{p} e^{-\left( \frac{(-k_x(x-(2-x_0)) + k_y(y-y_0) - ct)^2}{L^2} \right)}
$$

> 

Reviewer: ```suggestion

is the reflecting wave.

```
